### ISSUE TITLE
Handle "CodeAnalysisTreatWarningsAsErrors" MSBuild property for FxCop analyzers

### ISSUE THREAD

**Issue by mavasani** [dotnet/roslyn-analyzers#1887]
This issue tracks implementing the workaround mentioned by @sharwell in https://github.com/dotnet/roslyn/issues/16535#issuecomment-433947630:

--------------

_I discussed this with @mavasani offline and we seemed to reach the following conclusion:_

_Ideally, it will be easy to update the compiler to have a command line flag addressing this situation. I have no preference about the form or manner of implementation, so I leave that to @mavasani and the rest of the compiler team. Should problems with that approach be encountered, a viable workaround for the specific case of FXCop Analyzers is to generate a .targets file containing content like the following, and include the file in the FXCop Analyzer packages:_

```xml
<PropertyGroup Condition="'$(CodeAnalysisTreatWarningsAsErrors)' == 'false'">
  <WarningsNotAsErrors>$(WarningsNotAsErrors),CAxxxx,...</WarningsNotAsErrors>
</PropertyGroup>
```

-------
